server/src/main/scala/org/http4s/server/staticcontent/WebjarService.scala

```diff
@@ -0,0 +1,142 @@
+package org.http4s
+package server
+package staticcontent
+
+import java.util.concurrent.ExecutorService
+
+import scalaz.concurrent.{Strategy, Task}
+
+/**
+  * Constructs new services to serve assets from Webjars
+  */
+object WebjarService {
+
+  /** [[org.http4s.server.staticcontent.WebjarService]] configuration
+    *
+    * @param filter To filter which assets from the webjars should be served
+    * @param pathPrefix prefix of the Uri that content will be served from
+    * @param executor `ExecutorService` to use when collecting content
+    * @param cacheStrategy strategy to use for caching purposes. Default to no caching.
+    */
+  final case class Config(filter: WebjarAssetFilter = _ => true,
+                          pathPrefix: String = "",
+                          executor: ExecutorService = Strategy.DefaultExecutorService,
+                          cacheStrategy: CacheStrategy = NoopCacheStrategy)
+
+  /**
+    * Contains the information about an asset inside a webjar
+    *
+    * @param library The webjar's library name
+    * @param version The version of the webjar
+    * @param asset The asset name inside the webjar
+    */
+  final case class WebjarAsset(library: String, version: String, asset: String, pathPrefix: String) {
+
+    /**
+      * The path under which the asset is reachable in the web
+      *
+      * @return The path where the asset is reachable on the web
+      */
+    private lazy val webPath: String =
```

Comment:
It could be used in the callbacks to filter files - but it's just a convinience. I can remove it.